sgx-types/src/ssa.rs

```diff
@@ -270,45 +272,143 @@ pub struct Miscellaneous {
+        }
+    }
+}
+
+/// The XSAVE header of an XSAVE area comprises the 64 bytes starting at offset 512 from the area’s base address.
+/// See 13.4.2 in
+/// https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-1-manual.pdf
 #[derive(Debug)]
-#[repr(C, align(4096))]
-pub struct Footer {
-    /// Size == 4096 - 256
-    padding0: [[u64; 32]; 15],
-    /// 256 - size_of::<Miscellaneous>() - size_of::<Gpr>()
-    padding1: [u32; 14],
-    /// May hold some exception info for faults occurring inside an enclave.
-    pub misc: Miscellaneous,
-    /// Holds info on general purpose registers.
-    pub gpr: Gpr,
+#[repr(C)]
+pub struct XSaveHeader {
+    /// State-component bitmap identifying the state components in the XSAVE area
+    pub xstate_bv: u64,
+    /// State-component bitmap indicating the format of the XSAVE extended region and whether
+    /// a component is in the XSAVE area
+    pub xcomp_bv: u64,
+    reserved0: [u64; 6],
 }
 
-impl Footer {
-    /// Creates a new footer from Miscellaneous (fault-related) and Gpr info.
-    pub const fn new(misc: Miscellaneous, gpr: Gpr) -> Self {
+impl Default for XSaveHeader {
+    fn default() -> Self {
         Self {
-            padding0: [[0; 32]; 15],
-            padding1: [0; 14],
-            misc,
-            gpr,
+            xstate_bv: Default::default(),
+            /// XCOMP_BV[63] = 1, compaction mode for the XSave Extended Region. However, no state
+            /// components are included in the Extended Region as all other bits are 0. The size of
+            /// the XSaveXtd is therefore zero.
+            xcomp_bv: 0x80000000, // TODO: This is setting bit 31. Check on which is correct.
+            reserved0: Default::default(),
         }
     }
+}
 
+/// For our use case, the XSave Extended Region is in compacted format and currently holds nothing, as specifed in XCOMP_BV in
+/// https://github.com/jsakkine-intel/linux-sgx/blob/master/tools/testing/selftests/x86/sgx/encl_bootstrap.S#L89.
+/// For more on this region, see 13.4.3 in
+/// https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-1-manual.pdf.
+#[derive(Debug, Default)]
+#[repr(C)]
+pub struct XSaveXtd;
+
+/// For details on the fields included in XSave, see Section 13.4 in
+/// https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-1-manual.pdf.
+#[derive(Debug, Default)]
+#[repr(C)]
+pub struct XSave {
+    /// Legacy region of the XSave area
+    pub legacy: XSaveLegacy,
+    /// XSave header
+    pub header: XSaveHeader,
+    /// XSave Extended Region (not used)
+    pub xtd: XSaveXtd,
+}
+
+/// When an AEX occurs while running in an enclave, the architectural state is saved
+/// in the thread’s current StateSaveArea (SSA Frame), which is pointed to by TCS.CSSA.
+///
+/// Section 38.9, Table 38-7
+#[repr(C)]
```

Comment:
We probably do want an implementation of `Debug` here. Debugging the saved state will be common...

Reply:
This is going to take a bit longer, so I'd prefer to submit it as a separate issue/PR.